### ISSUE TITLE
Rule proposal: `prefer-prototype-methods`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1223]
### Fail

```js
[].push.apply(foo, elements);
```

```js
[].slice.call(foo);
```

```js
({}).toString.call(foo);
```

```js
({}).hasOwnProperty.call(foo, 'bar');
```

### Pass

```js
Array.prototype.push.apply(foo, elements);
```

```js
Array.prototype.slice.call(foo);
```

```js
Object.prototype.toString.call(foo);
```

```js
Object.prototype.hasOwnProperty.call(foo, 'bar');
```

Alter name `no-literal-prototype`

**Comment by fisker** [comment]
Maybe we should also fobid `foo.constructor.prototype.bar.call()`

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
These are all methods, I think better name is `prefer-prototype-methods`.
